src/main/java/org/opensrp/web/rest/ClientFormResource.java

```diff
@@ -0,0 +1,167 @@
+        DefaultArtifactVersion formVersionRequired = new DefaultArtifactVersion(formVersion);
+        DefaultArtifactVersion currentFormVersionV = null;
+        if (!TextUtils.isEmpty(currentFormVersion)) {
+            currentFormVersionV = new DefaultArtifactVersion(currentFormVersion);
+        }
+
+        if (currentFormVersionV != null && currentFormVersionV.compareTo(formVersionRequired) > 0) {
+            return new ResponseEntity<>((String) null, HttpStatus.BAD_REQUEST);
+        }
+
+        if (!clientFormService.isClientFormExists(formIdentifier)) {
+            return new ResponseEntity<>((String) null, HttpStatus.NOT_FOUND);
+        }
+
+        // Check if the form identifier with that version exists
+        ClientFormMetadata clientFormMetadata = clientFormService.getClientFormMetadataByIdentifierAndVersion(formIdentifier, formVersion);
+        ClientFormService.CompleteClientForm completeClientForm = null;
+
+        long formId;
+
+        if (clientFormMetadata == null) {
+            // Get an older form version
+            clientFormMetadata = getMostRecentVersion(formIdentifier);
+            if (clientFormMetadata == null) {
+                return new ResponseEntity<>((String) null, HttpStatus.NOT_FOUND);
+            } else {
+                formId = clientFormMetadata.getId();
+            }
+
+        } else {
+            formId = clientFormMetadata.getId();
+        }
+
+        ClientForm clientForm = clientFormService.getClientFormById(formId);
+        if (clientForm == null) {
+            return new ResponseEntity<>((String) null, HttpStatus.NOT_FOUND);
+        }
+
+        completeClientForm = new ClientFormService.CompleteClientForm(clientForm, clientFormMetadata);
+
+        return new ResponseEntity<>(objectMapper.writeValueAsString(completeClientForm), HttpStatus.OK);
+    }
+
+    private ClientFormMetadata getMostRecentVersion(@NonNull String formIdentifier) {
+        List<IdVersionTuple> availableFormVersions = clientFormService.getAvailableClientFormMetadataVersionByIdentifier(formIdentifier);
+        DefaultArtifactVersion highestVersion = null;
+        IdVersionTuple highestIdVersionTuple = null;
+
+        for (IdVersionTuple availableFormVersion : availableFormVersions) {
+            DefaultArtifactVersion semanticFormVersion = new DefaultArtifactVersion(availableFormVersion.getVersion());
+
+            if (highestVersion == null || semanticFormVersion.compareTo(highestVersion) > 0) {
+                highestVersion = semanticFormVersion;
+                highestIdVersionTuple = availableFormVersion;
+            }
+        }
+
+        if (highestVersion == null) {
+            return null;
+        } else {
+            long formId = highestIdVersionTuple.getId();
+            return clientFormService.getClientFormMetadataById(formId);
+        }
+    }
+
+    @RequestMapping(headers = {"Accept=multipart/form-data"}, method = RequestMethod.POST)
+    @ResponseBody
```

Comment:
This is not required. Its usually put in the method parameter to bind the body of the request to pojo or string